pandas/tests/series/methods/test_shift.py

```diff
@@ -344,3 +344,14 @@ def test_shift_preserve_freqstr(self, periods):
             index=pd.date_range("2016-1-1 02:00:00", periods=periods, freq="H"),
         )
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "input_value, expected_value",
+        [(np.empty(shape=(0,)), []), (np.ones(shape=(2,)), [np.nan, 1.0])],
+    )
+    def test_shift_non_writable_array(self, input_value, expected_value):
+        # GH21049 Verify whether non writable numpy array is shiftable
+        input_value.setflags(write=False)
+        tm.assert_series_equal(
+            pd.Series(input_value).shift(1), pd.Series(expected_value, dtype="float64")
```

Comment:
Could you create variables `result` and `expected` so this can read `tm.assert_series_equal(result, expected)`